python/lsst/ts/externalscripts/auxtel/build_pointing_model.py

```diff
@@ -0,0 +1,290 @@
+        )
+
+        self.elevation_grid = np.array(elevation[mask_out_of_range])
+        self.azimuth_grid = np.array(azimuth[mask_out_of_range])
+
+        self.log.debug("Sorting data in azimuth.")
+
+        azimuth_sort = np.argsort(self.azimuth_grid)
+
+        self.elevation_grid = self.elevation_grid[azimuth_sort]
+        self.azimuth_grid = self.azimuth_grid[azimuth_sort]
+
+        self.log.debug(f"Grid size: {self.grid_size}.")
+
+    @property
+    def grid_size(self):
+        return len(self.elevation_grid)
+
+    @property
+    def images_per_position(self):
+        return 2
+
+    @property
+    def camera_readout_time(self):
+        return 2.0
+
+    @property
+    def estimated_average_slew_time(self):
+        """The estimated average slew time considers a slew speed of 1 deg/sec.
+        With that in mind, it returs the resolution of the grid in degrees.
+        """
+        return hp.nside2resol(self.config.nside, arcmin=True) / 60.0
+
+    async def arun(self, checkpoint_active=False):
+
+        for grid_index, azimuth, elevation in zip(
+            range(self.grid_size), self.azimuth_grid, self.elevation_grid
+        ):
+
+            checkpoint_message = (
+                f"Execute grid position {grid_index+1} of {self.grid_size}: "
+                f"az={azimuth}, el={elevation}."
+            )
+
+            await self.handle_checkpoint(checkpoint_active, checkpoint_message)
+            await self.execute_grid(azimuth, elevation)
+
+    async def handle_checkpoint(self, checkpoint_active, checkpoint_message):
+
+        if checkpoint_active:
+            await self.checkpoint(checkpoint_message)
+        else:
+            self.log.info(checkpoint_message)
+
+    async def execute_grid(self, azimuth, elevation):
+        """Execute one grid position.
+
+        Parameters
+        ----------
+        azimuth : `float`
+            Azimuth of the grid position (in degrees).
+        elevation : `float`
+            Elevation of the grid position (in degrees).
+        """
+
+        try:
+            target = await self.atcs.find_target(
+                az=azimuth, el=elevation, mag_limit=self.config.magnitude_limit
+            )
+        except Exception:
+            self.log.exception(
+                f"Error finding target for azimuth={azimuth}, elevation={elevation}."
+                "Skipping grid position."
+            )
+            self.iterations["failed"] += 1
+            return
+
+        await self.atcs.slew_object(name=target, rot_type=RotType.PhysicalSky)
+
+        await self.center_on_brightest_source()
+
+        self.iterations["successful"] += 1
+
+    async def center_on_brightest_source(self):
+
+        self.latiss.rem.atarchiver.evt_imageInOODS.flush()
+
+        acquisition_image_ids = await self.latiss.take_engtest(
+            exptime=self.config.exposure_time, n=1
+        )
+
+        await self.latiss.rem.atarchiver.evt_imageInOODS.next(
```

Comment:
there is technically a race condition here.. but it probably doesn't matter.
i suppose the proper thing to do is verify the imageInOODS event matches the acquisition_image_ids

Reply:
uhm... I see.. maybe we should add something to `LATISS` to deal with that.. I've seen this more than once and you are right, we may want to check that the `imageInOODS` event is for the correct image id.

Reply:
So the idea was to refactor this stuff when we switch to gen3... which was supposed to be this week...